src/com/sun/jna/NativeLibrary.java

```diff
@@ -155,27 +155,27 @@ private static NativeLibrary loadLibrary(String libraryName, Map<String, ?> opti
 
         List<Throwable> exceptions = new ArrayList<Throwable>();
         boolean isAbsolutePath = new File(libraryName).isAbsolute();
-        List<String> searchPath = new ArrayList<String>();
+        LinkedHashSet<String> searchPath = new LinkedHashSet<String>();
         int openFlags = openFlags(options);
 
-        // Append web start path, if available.  Note that this does not
-        // attempt any library name variations
-        String webstartPath = Native.getWebStartLibraryPath(libraryName);
-        if (webstartPath != null) {
-            LOG.log(DEBUG_LOAD_LEVEL, "Adding web start path " + webstartPath);
-            searchPath.add(webstartPath);
-        }
-
         //
         // Prepend any custom search paths specifically for this library
         //
         List<String> customPaths = searchPaths.get(libraryName);
         if (customPaths != null) {
             synchronized (customPaths) {
-                searchPath.addAll(0, customPaths);
+                searchPath.addAll(customPaths);
```

Comment:
As there are no 'prepend' in LinkedHashSet, the custom paths are moved forward to keep the order.